### PR TITLE
@metamask/eslint config@6.0.0

### PR DESCRIPTION
**Pull request by rekmarks**
This PR adds `@metamask/eslint-config@6.0.0` and sibling packages from that major version. As you can see, the diff is enormous. However, most of the changes are programmatic and contained within single commits, so reviewing won't be as bad as it looks.

The end result of this effort is to bring this repository onto the same lint standards as everywhere else in the organization. The package's own `prettier` config is removed.

Changes prefixed with "❗" required manual fixing and deserve extra attention.

- 05500ab
  - This makes the necessary package updates and changes to `.eslintrc.js`
  - ❗ Notably, it stops extending the `jest` rules for every file, and moves them into an override targeting test files only.
  - ❗ Redundant rules were removed in this commit. 
    - "Redundant" meaning they were identically configured by an extended config.
- b99cdb0
  - ❗ Update the CI config and package scripts. The `format` scripts and CI job are removed since formatting is now done during `lint`.
- b8fa351
  - Run `yarn lint:fix` and replace `toEqual` with `toStrictEqual` in tests.
  - Some instances of `toEqual` were left over after the rebase. They're fixed fdec723.
- 2fa8725
  - ❗ Manually fix test failures resulting from the migration to `toStrictEqual`.
- 257fc97
  - ❗ Manually replace `jest` `.resolves` matchers.
- 5f3d819
  - ❗ Update local `jest` rules and move them to the override added in 5b4957c.
- e5090fb
  - Remove `lint-staged` _and_ the local `prettier` config.
- fdec723
  - Run `yarn lint:fix` to fix all prettier violations following the removal of the local `prettier` config.
  - Replace remaining `toEqual` instances with `toStrictEqual`.

### CODE REVIEW

**Review by ryanml — APPROVED** [review]
Config changes look good and seem to be in line with recent changes across the organization. Programmatic diffs look fine as well. I'll additionally refer to @Gudahtt for review

**Comment by rekmarks** [comment]
`/yolo`

**Comment by Gudahtt** [comment]
I'm tempted to ask that this be delayed until after #378 and #387. Both of those PRs would have many conflicts with this, and I suspect it would be easier to resolve them in this PR because the last commit can be redone fairly easily (the `yarn lint:fix` one).

File: .eslintrc.js [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
+    {
+      files: ['*.test.ts', '*.test.js'],
+      extends: ['@metamask/eslint-config-jest'],
+      rules: {
+        'jest/no-conditional-expect': 'off',
+        'jest/no-restricted-matchers': [
+          'error',
+          {
+            resolves: 'Use `expect(await promise)` instead.',
+            // 'toBeFalsy': 'Avoid `toBeFalsy`',
```
Comment (Gudahtt) [inline .eslintrc.js:23]: :eyes:

Comment (Gudahtt) [reply .eslintrc.js:23]: Ah I see, these are restricted matchers in our config that we use here? Maybe worth adding a comment explaining this

File: .eslintrc.js [hunk 2]
```diff
@@ -1,26 +1,50 @@
...
+    'docs',
+    'coverage',
+    '*.d.ts',
+  ],
   overrides: [
+    {
+      files: ['*.test.ts', '*.test.js'],
+      extends: ['@metamask/eslint-config-jest'],
+      rules: {
+        'jest/no-conditional-expect': 'off',
```
Comment (Gudahtt) [inline .eslintrc.js:18]: Maybe worth copying the `// TODO: re-enble most of these rules` comment here too.

File: .eslintrc.js [hunk 3]
```diff
@@ -1,26 +1,50 @@
...
         sourceType: 'script',
       },
     },
     {
       files: ['*.ts'],
-      extends: ['@metamask/eslint-config/config/typescript'],
+      extends: ['@metamask/eslint-config-typescript'],
       rules: {
         // `no-shadow` has incompatibilities with TypeScript
+        // TODO: Migrate this into @metamask/eslint-config?
```
Comment (Gudahtt) [inline .eslintrc.js:44]: Yes, I think we should do this.

```suggestion
        // TODO: Migrate this into @metamask/eslint-config
```

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**